modules/table/src/integrationTest/java/org/apache/ignite/internal/table/distributed/disaster/AbstractHighAvailablePartitionsRecoveryTest.java

```diff
@@ -132,10 +142,87 @@ private void waitAndAssertStableAssignmentsOfPartitionEqualTo(
                 .collect(Collectors.toUnmodifiableSet()));
     }
 
+    /**
+     * Wait for the 2 facts simultaneously.
+     *
+     * <ul>
+     *     <li>All planned rebalances have finished (pending and planned keys is empty).</li>
+     *     <li>Stable assignments is equal to expected</li>
+     * </ul>
+     *
+     * @param gatewayNode Node for communication with cluster and components.
+     * @param tableName Table name.
+     * @param partitionIds Set of target partition ids to check.
+     * @param nodes Expected set of nodes in stable assignments.
+     */
+    final void waitThatAllRebalancesHaveFinishedAndStableAssignmentsEqualsToExpected(
+            IgniteImpl gatewayNode, String tableName, Set<Integer> partitionIds, Set<String> nodes) {
+        partitionIds.forEach(p -> {
+            try {
+                waitThatAllRebalancesHaveFinishedAndStableAssignmentsEqualsToExpected(gatewayNode, tableName, p, nodes);
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+        });
+    }
+
+    /**
+     * The same as the previous one, but for concrete partition.
```

Comment:
Let's put link to the original method

Reply:
I can't ( The link is + ** is longer than 144 symbols. I tried when was making these changes) 
<img width="1559" alt="Screenshot 2025-01-23 at 16 45 42" src="https://github.com/user-attachments/assets/bbf2155c-0245-412c-bf60-e7899574afab" />

Reply:
Check this solution pls https://github.com/apache/ignite-3/pull/5064/commits/5ae9304e28d380241f78e8d330bc26a73e91d1dd